themes/gatsby-theme-blog/gatsby-config.js

```diff
@@ -92,30 +70,26 @@ module.exports = {
             },
           },
         ],
+        remarkPlugins: [
+          require(`remark-slug`),
+          [
+            require(`remark-autolink-headings`),
+            {
+              content: {
+                type: `element`,
+                // custom MDX element - is this too magical??
+                tagName: `link-icon`,
```

Comment:
By default `remark-autolink-headings` renders a `<span>` with a className. Here this is making a custom MDX element type, which is rendered by a custom MDX component that renders the Octicon link icon SVG. While this works, this seems a little too magical to me.

One alternative would be to use `remark-slug` and custom `h1` –`h6` MDX components that take `props.id` to render the link and icon – this would also open up other rendering options when shadowing the headings

Reply:
This strikes me as a bit too magical, partially because (and this comment may be borne of ignorance) but if the way the mdx plugin is handling it is different from the way it's handled in remark, isn't that something that should be fixed in the mdx plugin?

Reply:
MDX isn't doing anything non-standard here with the plugin, but I am with the `content` option here – and I definitely don't like this anymore

I think I'm leaning towards using custom MDX components for this sort of thing – when the markup generation happens in a Remark plugin, it's hidden away, whereas a React component would be easy to read and also shadow